packages/expo-camera/ios/EXCamera/EXCamera.m

```diff
@@ -523,25 +509,60 @@ - (void)record:(NSDictionary *)options resolve:(EXPromiseResolveBlock)resolve re
     }
     [connection setVideoOrientation:[EXCameraUtils videoOrientationForDeviceOrientation:[[UIDevice currentDevice] orientation]]];
     
-    AVCaptureSessionPreset preset;
-    if (options[@"quality"]) {
-      EXCameraVideoResolution resolution = [options[@"quality"] integerValue];
-      preset = [EXCameraUtils captureSessionPresetForVideoResolution:resolution];
-    } else if ([self.session.sessionPreset isEqual:AVCaptureSessionPresetPhoto]) {
-      preset = AVCaptureSessionPresetHigh;
-    }
-
-    if (preset != nil) {
-      [self updateSessionPreset:preset];
-    }
-    
     [self setVideoOptions:options forConnection:connection onReject:reject];
     
     bool canBeMirrored = connection.isVideoMirroringSupported;
     bool shouldBeMirrored = options[@"mirror"] && [options[@"mirror"] boolValue];
     if (canBeMirrored && shouldBeMirrored) {
       [connection setVideoMirrored:shouldBeMirrored];
     }
+}
+
+-(void) updateSessionWithOptions:(NSDictionary *)options
+{
+  bool shouldBeMuted = options[@"mute"] && [options[@"mute"] boolValue];
+  [self updateSessionAudioIsMuted:shouldBeMuted];
+
+  
+  AVCaptureSessionPreset preset;
+  if (options[@"quality"]) {
+    EXCameraVideoResolution resolution = [options[@"quality"] integerValue];
+    preset = [EXCameraUtils captureSessionPresetForVideoResolution:resolution];
+  } else if ([self.session.sessionPreset isEqual:AVCaptureSessionPresetPhoto]) {
+    preset = AVCaptureSessionPresetHigh;
+  }
+
+  if (preset != nil) {
+    [self updateSessionPreset:preset];
+  }
+}
+
+- (void)record:(NSDictionary *)options resolve:(EXPromiseResolveBlock)resolve reject:(EXPromiseRejectBlock)reject
+{
+  if (_movieFileOutput == nil) {
+    // At the time of writing AVCaptureMovieFileOutput and AVCaptureVideoDataOutput (> GMVDataOutput)
+    // cannot coexist on the same AVSession (see: https://stackoverflow.com/a/4986032/1123156).
+    // We stop face detection here and restart it in when AVCaptureMovieFileOutput finishes recording.
+    if (_faceDetectorManager) {
+      [_faceDetectorManager stopFaceDetection];
+    }
+    [self setupMovieFileCapture];
+  }
+
+    
+  if (_movieFileOutput != nil && !_movieFileOutput.isRecording && _videoRecordedResolve == nil && _videoRecordedReject == nil) {
+    // there are some options, which are called within commitConfiguration. These kind of options need to
+    // be preset as early as possible, however, if we receive a new options - we still configure them.
+    // All these options are configured in `updateSessionWithOptions`
+    // However, some fields should be configured when starting a video because it depends on movieFileOutput
+    // these options don't cause flickering => we can configure it before starting a video
+    // we do it in `upadteVideoWithOptions`
```

Comment:
```suggestion
    // However, some fields should be configured when starting a video because it depends on `movieFileOutput`.
    // These options don't cause flickering => we can configure it before starting recording - we do it in `upadteVideoWithOptions`.
```